universal-login-contracts/test/contracts/Wallet/WalletImplementation.ts

```diff
@@ -30,18 +26,25 @@ describe('WalletImplementation', () => {
     ({provider, publicResolver, registrarAddress, ensAddress, wallet} = await loadFixture(basicENS));
     [, , , otherWallet] = getWallets(provider);
     ensArgs = [hashLabel, name, node, ensAddress, registrarAddress, publicResolver];
+    const transactionCount = await wallet.getTransactionCount();
+    const futureAddress = computeContractAddress(wallet.address, transactionCount);
+    await otherWallet.sendTransaction({to: futureAddress, value: utils.parseEther('1')});
   });
 
   it('wallet implementation deployed and initialized successfully', async () => {
-    const transactionCount = await wallet.getTransactionCount();
-    const futureAddress = computeCounterfactualAddress(wallet.address, transactionCount);
-    await otherWallet.sendTransaction({to: futureAddress, value: utils.parseEther('1')});
     const walletContract = await deployContract(wallet, WalletContract, [], {gasLimit: 5000000});
 
     await walletContract.initializeWithENS(keyPair.publicKey, ...ensArgs, TEST_GAS_PRICE, ETHER_NATIVE_TOKEN.address);
     expect(utils.formatEther(await provider.getBalance(walletContract.address))).to.eq('0.999999999999316');
   });
 
+  it('refund after initialize', async () => {
+    const walletContract = await deployContract(wallet, WalletContract, [], {gasLimit: 5000000});
```

Comment:
Creating walletContract can be moved to `beforeEach`

Reply:
I would add tests, to have 3 options:
'Can't initalize twice'
'Can't initalize twice with ENS'
'Can't initalize twice with and without ENS'

Reply:
✅